adapters/logstash/logstash.go

```diff
@@ -0,0 +1,63 @@
+package logstash
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"os"
+
+	"github.com/gliderlabs/logspout/router"
+)
+
+var hostname string
+
+func init() {
+	router.AdapterFactories.Register(NewLogstashAdapter, "logstash")
+
+	hostname, _ = os.Hostname()
+}
+
+// LogstashAdapter is an adapter that streams UPD JSON to Logstash.
+type LogstashAdapter struct {
+	conn  net.Conn
+	route *router.Route
+}
+
+// NewLogstashAdapter creates a LogstashAdapter with UDP transport.
+func NewLogstashAdapter(route *router.Route) (router.LogAdapter, error) {
+	transport, found := router.AdapterTransports.Lookup(route.AdapterTransport("udp"))
+	if !found {
+		return nil, errors.New("unable to find adapter: " + route.Adapter)
+	}
+
+	conn, err := transport.Dial(route.Address, route.Options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &LogstashAdapter{
+		route: route,
+		conn:  conn,
+	}, nil
+}
+
+func (adapter *LogstashAdapter) Stream(logstream chan *router.Message) {
+	for m := range logstream {
+		msg := LogstashMessage{
+			Time:     m.Time.Unix(),
+			Message:  m.Data,
+			Hostname: hostname,
+			Image:    m.Container.Config.Image,
```

Comment:
It seems like this adapter is not as general as you think. Maybe it deserves to live as your own third-party module? 

I say this because if I were to do a general Logstash adapter, I probably wouldn't use JSON+UDP which is just one possible input format ... If I were to pick a canonical Logstash adapter machanism, I'd use the [logstash-forwarder](https://github.com/elastic/logstash-forwarder) protocol, perhaps even use it directly as a library. 

Don't you agree?